### PR TITLE
TRT-1463: use bigquery to get releases for component readiness

### PR DESCRIPTION
**Pull request by xueqzhan**

### CODE REVIEW

**Comment by stbenjam** [comment]
This is about 4-5x slower - could we cache it? Could be as simple as wrapping this in the cache middleware here: https://github.com/openshift/sippy/blob/master/pkg/sippyserver/server.go#L1347

Postgres:
```
$ seq 1 200 | xargs -I{} -n1 curl -o /dev/null -s -w '%{time_total}\n' http://localhost:8080/api/releases | awk '{sum+=$1} END {print "Average total response time: " sum/NR " seconds"}'

Average total response time: 0.122231 seconds
```

Bigquery:

```
$ seq 1 200 | xargs -I{} -n1 curl -o /dev/null -s -w '%{time_total}\n' http://localhost:8080/api/releases | awk '{sum+=$1} END {print "Average total response time: " sum/NR " seconds"}'

Average total response time: 0.5315 seconds
```

File: pkg/sippyserver/server.go [hunk 1]
```diff
@@ -866,18 +866,21 @@ func (s *Server) jsonReleasesReportFromDB(w http.ResponseWriter, _ *http.Request
...
 	}
 
 	for _, release := range releases {
 		response.Releases = append(response.Releases, release.Release)
+		if release.GADate != nil {
+			response.GADates[release.Release] = *release.GADate
+		}
 	}
 
 	type LastUpdated struct {
```
Comment (xueqzhan) [inline pkg/sippyserver/server.go:886]: Current change is not totally independent from postgres due to the last updated query below. Should this be also modified?

Comment (stbenjam) [reply pkg/sippyserver/server.go:886]: How about just not specifying it if there's no local db?   https://github.com/openshift/sippy/pull/1559/files#diff-768a5e234f25507b29e6156fd5af2bfb561aef7c7b19612e7e73ea4cb6be07e4R913

That's how I solved it for now, last updated is really only relevant to the data in postgres anyway

Comment (xueqzhan) [reply pkg/sippyserver/server.go]: I added the equivalent from BQ

File: pkg/api/utils.go [hunk 2]
```diff
@@ -103,3 +103,27 @@ func GetReleases(dbc *db.DB, bqc *bqclient.Client) ([]query.Release, error) {
 	}
 	return query.ReleasesFromDB(dbc)
 }
+
+// GetLastUpdated gets the update time of the latest prow job run if bqc is defined.
+// Otherwise, it falls back to get it from sippy DB
+func GetLastUpdated(dbc *db.DB, bqc *bqclient.Client) (time.Time, error) {
```
[the remark this answers is not in the document]
Comment (xueqzhan) [reply pkg/api/utils.go:109]: But what are we displaying on the Sippy top right if we do not have a value?

[the remark this answers is not in the document]
Comment (xueqzhan) [reply pkg/api/utils.go:109]: I tried it: Last updated 738971 days ago

**Comment by xueqzhan** [comment]
Latest commit skipped the last updated fetching the same way you did. @stbenjam

File: pkg/api/utils.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (stbenjam) [reply pkg/api/utils.go]: > But what are we displaying on the Sippy top right if we do not have a value?

I am not showing a value when there's no postgres client.

[the remark this answers is not in the document]
Comment (stbenjam) [reply pkg/api/utils.go]: Our main sippy always has a bigquery client, so you're going to show the value from the jobs table, and we're going to lose our ability to see when Sippy's DB last updated.  It's what our last updated alerts are based on.  In my mind "last updated" **only** applies to the local db.

**Pull request merged**